localstack/services/awslambda/invocation/executor_endpoint.py

```diff
@@ -71,26 +70,51 @@ def invocation_logs(invoke_id: str) -> ResponseReturnValue:
                 # TODO handle error in some way?
             return Response(status=HTTPStatus.ACCEPTED)
 
-        @executor_endpoint.route("/status/<executor_id>/ready", methods=["POST"])
-        def status_ready(executor_id: str) -> ResponseReturnValue:
+        def status_ready(request: Request, executor_id: str) -> Response:
             self.service_endpoint.status_ready(executor_id=executor_id)
             return Response(status=HTTPStatus.ACCEPTED)
 
-        @executor_endpoint.route("/status/<executor_id>/error", methods=["POST"])
-        def status_error(executor_id: str) -> ResponseReturnValue:
+        def status_error(request: Request, executor_id: str) -> Response:
             self.service_endpoint.status_error(executor_id=executor_id)
             return Response(status=HTTPStatus.ACCEPTED)
 
-        return executor_endpoint
-
-    def do_run(self) -> None:
-        endpoint = self._create_endpoint()
-        LOG.debug("Running executor endpoint API on %s:%s", self.host, self.port)
-        endpoint.run(self.host, self.port)
-
-    def do_shutdown(self) -> None:
-        if self._thread:
-            self._thread.stop()
+        return [
+            router.add(
+                f"{self.get_endpoint_prefix()}/invocations/<req_id>/response",
+                endpoint=invocation_response,
+                methods=["POST"],
+            ),
+            router.add(
+                f"{self.get_endpoint_prefix()}/invocations/<req_id>/error",
+                endpoint=invocation_error,
+                methods=["POST"],
+            ),
+            router.add(
+                f"{self.get_endpoint_prefix()}/invocations/<invoke_id>/logs",
+                endpoint=invocation_logs,
+                methods=["POST"],
+            ),
+            router.add(
+                f"{self.get_endpoint_prefix()}/status/<executor_id>/ready",
+                endpoint=status_ready,
+                methods=["POST"],
+            ),
+            router.add(
+                f"{self.get_endpoint_prefix()}/status/<executor_id>/error",
+                endpoint=status_error,
+                methods=["POST"],
+            ),
+        ]
+
+    def get_endpoint_prefix(self):
+        return f"{NAMESPACE}/{self.endpoint_id}"
+
+    def start(self) -> None:
+        self.rules = self._create_endpoint(self.router)
```

Comment:
Agree, will fix this in a future iteration. Was initially meant to provide the router as a parameter instead of using global, now it is a object attribute anyway.